src/SignalsHandler.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace React\EventLoop;
+
+use React\EventLoop\Timer\TimerInterface;
+
+/**
+ * @internal
+ */
+final class SignalsHandler
+{
+    private $loop;
+    private $timer;
+    private $signals = [];
+    private $on;
+    private $off;
+
+    public function __construct(LoopInterface $loop, callable $on, callable $off)
+    {
+        $this->loop = $loop;
+        $this->on = $on;
+        $this->off = $off;
+    }
+
+    public function __destruct()
+    {
+        $off = $this->off;
+        foreach ($this->signals as $signal => $listeners) {
+            $off($signal);
+        }
+    }
+
+    public function add($signal, callable $listener)
+    {
+        if (count($this->signals) == 0 && $this->timer === null) {
+            $this->timer = $this->loop->addPeriodicTimer(1, function () {});
+        }
```

Comment:
I would suggest using a much higher interval do prevent the CPU from waking up every second. Also, maybe add a simple comment what this does?

Reply:
Don't really see the harm in once a second but increased it to five minutes